### ISSUE TITLE
Incorrect button mappings for Casio fx-9860GII SD

### ISSUE THREAD

**Issue by GinkREAL** [voidware/reckon#1]
I made an account just for this lol, anyway the buttons are mapped incorrectly, like 3 is delete, 9 is sin and so on.

**Comment by voidware** [comment]
Yes, the original code was for the 9860 slim not the GII. If you know the mappings for this model, that would be great. I never got to test the GII. thanks,

**Comment by GinkREAL** [comment]
yeah sure i can test it out. How would i do that though?

**Comment by voidware** [comment]
Sorry, missing file rpn.cpp added.

**Comment by wisemoth** [comment]
It's taken a while and fixes to get the source to build; now I've hacked a trial for the key map on the non-slim model.  The problem is the current code for testing if the calculator is a slim is failing.  If you force the `IsSlim` to return `0` then the keymap already in the code for the "Portrait" calculator works well on the non-slim calculator :)

Edit: Though some that look like the map should work, don't quite - eg `Abs(` (and others) appear as `u`

**Comment by 2minwia** [comment]
pls fix, 10 bux for fixer
https://www.bountysource.com/issues/33052300-incorrect-button-mappings-for-casio-fx-9860gii-sd

**Comment by watchmakerful** [comment]
Any news on this issue? The keymap on a FX-9750GII with a Graph75+E firmware (similar to FX-9860Gii) is also incorrect.
PS, I can't build this app with the Casio SDK, it can't find some source files even if they actually exist.

**Comment by watchmakerful** [comment]
Trying the latest build with IsSlim forced to return 0. Casio FX-9750GII flashed into Graph75+E v2.05.
Some buttons have problems:
Exit (u) 
F2 (u)
F6 (u)
Shift+X0t (space) !!!
Shift+ab/c (space) !!!
Shift+f-d (h)
Shift+ -> (u)
Shift+1, 4, 7, 9 (u)
Shift+5 (lowercase q)
Shift+6 (lowercase r) (are these two by design???)
Shift+0 (space) !!!

**Comment by watchmakerful** [comment]
PS, plotting functions such as X^2, X^3 makes my calculator immediately reboot.

**Comment by watchmakerful** [comment]
One more problem, arrow keys in ALG mode also type "u".

**Comment by voidware** [comment]
It does that here too. I think there are no bindings for arrows in ALG 
mode. "u" seems to be the default. Maybe make it map to something.

Also, I seem to have lost my FA-124 PC LINK program. so i can't copy to 
a real device. My fixes are in the emulator. is there a download link 
for FA-124 (or a new version). Also it needs to work on Windows 64 bit. 
I think i originally had it installed on an old machine.

-- hugh.

**Comment by watchmakerful** [comment]
Download here: http://edu.casio.com/forteachers/er/software/
or directly from my Google drive: https://drive.google.com/open?id=1hSxeC7bdzjuKwABCCsgDb4GzBrIHUxFZ

**Comment by LdBeth** [comment]
> Some buttons have problems:

this is due to the `GetLine` function in `Reckon.c` get function keys that have not been assigned operations handled as normal alphabets, then inserted as garbage (they're actually not `u`, but been displayed so due to the internal character encoding). I think I can add guards to disable these keys, since `DEL` won't properly erase them, which is annoying, while the RPN mode do has implemented guards.
